overlord/state/task.go

```diff
@@ -485,8 +485,12 @@ func NewTaskSet(tasks ...*Task) *TaskSet {
 }
 
 // Edge returns the task marked with the given edge name.
-func (ts TaskSet) Edge(e TaskSetEdge) *Task {
-	return ts.edges[e]
+func (ts TaskSet) Edge(e TaskSetEdge) (*Task, error) {
+	t, ok := ts.edges[e]
+	if !ok {
+		return nil, ErrNoState
```

Comment:
ErrNoState doesn't carry a lot of information, but the usage of Edge will be mostly static (afawk) so I think

   fmt.Errorf("internal error: no %q edge in task set", e)  

would be maybe better

Reply:
Probably slight clearer  with "missing" instead of "no"